pkg/splunk/enterprise/configuration_test.go

```diff
@@ -236,3 +236,229 @@ func TestSetVolumeDefault(t *testing.T) {
 		}
 	}
 }
+
+func TestSmartstoreApplyIndexerClusterFailsOnInvalidSmartStoreConfi(t *testing.T) {
+	cr := enterprisev1.IndexerCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "standalone",
+			Namespace: "test",
+		},
+		Spec: enterprisev1.IndexerClusterSpec{
+			Replicas: 1,
+			SmartStore: enterprisev1.SmartStoreSpec{
+				VolList: []enterprisev1.VolumeSpec{
+					{Name: "msos_s2s3_vol", S3Endpoint: ""},
```

Comment:
So this test case should fail because `S3Endpoint: ""` ?

Reply:
Yes, in this case, its S3 end point. Point is to have some invaid smartstore config. Its tough to get in to all the combinations.